app/src/main/java/com/isc/hermes/MainActivity.java

```diff
@@ -24,7 +32,8 @@ public class MainActivity extends AppCompatActivity {
     private MapDisplay mapDisplay;
     private String mapStyle = "default";
     private CurrentLocationController currentLocationController;
-
+    private boolean visibilityMenu = false;
+    private GoogleAuthentication googleAuthentication;
```

Comment:
In this class, there should not be an instance of GoogleAuthentication because it is possible that the authentication types will be increased and this would generate many unnecessary instances in this class.

Reply:
Done